src/main/riddl/Members/members.riddl

```diff
@@ -157,24 +211,126 @@ context MembersContext  is {
+                    and "set entity lastModifiedOn to today()"
+                    and "yield event MemberStatusUpdated"
+                  }
+                  example ActiveMemberToTerminatedMember is {
+                    given "entity is transitioning from Active to Terminated"
                     then "set entity Status to Status field from message"
                     and "set entity lastModifiedBy to actingMember from message"
                     and "set entity lastModifiedOn to today()"
                     and "yield event MemberStatusUpdated"
+                  }
+                  example ActiveMemberIllegalStateTransition is {
+                    given "entity is transitioning from Active to Draft or Active"
+                    then error "Active members may not transition to desired state."
+                  }
                 }
                 on query GetMemberData {
                     then "yield Info for entity"
                 }
             }
         }
 
+        state Inactive is {
+          fields {
+            info: MembersContext.Info,
+            meta: MembersContext.MetaInfo
+          }
+
+          handler InactiveMemberHandler {
+            on command RegisterMember {then error "Member has already been registered."}
+            on command UpdateMemberStatus {
+              example InactiveMemberToActiveMember is {
+                given "entity is transitioning from Inactive to Active and required fields are non-empty"
+                then "set entity Status to Status field from message"
+                and "set entity lastModifiedBy to actingMember from message"
+                and "set entity lastModifiedOn to today()"
+                and "yield event MemberStatusUpdated"
+              }
+              example InactiveMemberToTerminatedMember is {
+                given "entity is transitioning from Inactive to Terminated and required fields are non-empty"
+                then "set entity Status to Status field from message"
+                and "set entity lastModifiedBy to actingMember from message"
+                and "set entity lastModifiedOn to today()"
+                and "yield event MemberStatusUpdated"
+              }
+              example InactiveMemberIllegalStateTransition is {
+                given "entity is transitioning from Draft to Draft, Inactive, or Suspended"
+                then error "Draft members may not transition to desired state."
+              }
+            }
+            on command UpdateMemberInfo {
+              then "set entity Info to non-Empty fields in UpdateInfo from message"
+              and "set entity lastModifiedBy to actingMember from message"
+              and "set entity lastModifiedOn to today()"
+              and "yield event MemberInfoUpdated"
+            }
+            on query GetMemberData {
+              then "yield Info for entity"
+            }
+          }
+        }
+
+        state Suspended is {
+          fields {
```

Comment:
I think we should add:
reason: String
as a required field for the suspended state. In other words, you cannot suspend a Member without explaining the action. So when someone attempts to reactivate the member they can determine if there has been proper resolution to the reason for suspension.